openlibrary/components/MergeUI/utils.js

```diff
@@ -230,6 +230,40 @@ function save_many(items, comment, action, data) {
     });
 }
 
+/**
+ * Fetches name associated with the author key
+ * @param {Object[]} works
+ * @returns {Promise<Record<string,object>} A response to the request
+ */
+export async function get_author_names(works) {
+    const authorIds = _.uniq(works).flatMap(record =>
+        (record.authors || []).map(authorEntry => authorEntry.author.key)
+    )
+
```

Comment:
Another place we can handle the case of no authors to avoid the network request.

```suggestion

    if (!authorIds.length) return {};

```